src/platforms/esp32/main/spi_driver.c

```diff
@@ -362,6 +370,153 @@ static term spidriver_write_at(Context *ctx, term req)
     return make_read_result_tuple(read_value, ctx);
 }
 
+static term populate_transaction(Context *ctx, struct spi_transaction_t *transaction, term transaction_term, bool write_only, size_t *output_size)
+{
+    term zero_term = term_from_int(0);
+
+    term command_term = term_get_map_assoc_default(ctx, transaction_term, context_make_atom(ctx, command_atom), zero_term);
+    if (!term_is_integer(command_term)) {
+        ESP_LOGE(TAG, "command transaction entry is not an integer");
+        return BADARG_ATOM;
+    }
+    transaction->cmd = (uint16_t) term_to_int(command_term);
+
+    term address_term = term_get_map_assoc_default(ctx, transaction_term, context_make_atom(ctx, address_atom), zero_term);
```

Comment:
we might accept `term_is_any_integer`, ad boxed integers, so we can use nearly any address.